haystack_experimental/components/preprocessors/embedding_based_document_splitter.py

```diff
@@ -0,0 +1,402 @@
+
+    def warm_up(self):
+        """
+        Warm up the component by initializing the sentence splitter.
+        """
+        nltk_imports.check()
+        self.sentence_splitter = SentenceSplitter(
+            language=self.language,
+            use_split_rules=self.use_split_rules,
+            extend_abbreviations=self.extend_abbreviations,
+            keep_white_spaces=True,
+        )
+        if hasattr(self.document_embedder, "warm_up"):
+            self.document_embedder.warm_up()
+        self._is_warmed_up = True
+
+    @component.output_types(documents=List[Document])
+    def run(self, documents: List[Document]) -> Dict[str, List[Document]]:
+        """
+        Split documents based on embedding similarity.
+
+        :param documents: The documents to split.
+        :returns: A dictionary with the following key:
+            - `documents`: List of documents with the split texts. Each document includes:
+                - A metadata field `source_id` to track the original document.
+                - A metadata field `split_id` to track the split number.
+                - A metadata field `page_number` to track the original page number.
+                - All other metadata copied from the original document.
+
+        :raises:
+            - `RuntimeError`: If the component wasn't warmed up.
+            - `TypeError`: If the input is not a list of Documents.
+            - `ValueError`: If the document content is None or empty.
+        """
+        if not self._is_warmed_up:
+            raise RuntimeError(
+                "The component EmbeddingBasedDocumentSplitter wasn't warmed up. Run 'warm_up()' before calling 'run()'."
+            )
+
+        if not isinstance(documents, list) or (documents and not isinstance(documents[0], Document)):
+            raise TypeError("EmbeddingBasedDocumentSplitter expects a List of Documents as input.")
+
+        split_docs: List[Document] = []
+        for doc in documents:
+            if doc.content is None:
+                raise ValueError(
+                    f"EmbeddingBasedDocumentSplitter only works with text documents but content for "
+                    f"document ID {doc.id} is None."
+                )
+            if doc.content == "":
+                logger.warning("Document ID {doc_id} has an empty content. Skipping this document.", doc_id=doc.id)
+                continue
+
+            doc_splits = self._split_document(doc)
+            split_docs.extend(doc_splits)
+
+        return {"documents": split_docs}
+
+    def _split_document(self, doc: Document) -> List[Document]:
+        """
+        Split a single document based on embedding similarity.
+        """
+        if self.sentence_splitter is None:
+            raise RuntimeError("Sentence splitter is not initialized. Call warm_up() first.")
+        if doc.content is None:
+            raise ValueError("Document content cannot be None.")
```

Comment:
Not entirely sure if these checks are needed since these are checked for already in the `run` method before this is called. Especially since this is a private method I'd be okay with removing these checks, but also fine to leave.